content/java/concepts/strings/terms/offsetByCodePoints/offsetByCodePoints.md

```diff
@@ -0,0 +1,98 @@
+---
+Title: '.offsetByCodePoints()'
+Description: 'Returns the new index of a character in a string after applying the specified code point offset.'
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'Strings'
+  - 'Methods'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **`offsetByCodePoints()`** function is used to calculate the index of a character in a string by specifying a starting index and a code point offset. It returns the new index after applying the offset limit.
+
+It is useful when working with strings that contain characters with multiple code points, such as emojis or characters from non-Latin scripts. This function helps accurately navigate and manipulate the index based on code points, allowing you to access specific characters or portions of the string.
+
+## Syntax
+
+```java
+Strings.offsetByCodePoints(int startIndex,int  offsetIndex)
+```
+
+- `startIndex` (int): The starting index in the string from which the offset is applied.
+
+- `OffsetIndex` (int): The offset, in terms of code points, by which the index is adjusted. Positive values move the index forward, and negative values move it backward.
+
+## Example 1
+
+In this example, the code point offset is set to 7, which means the new index will be 7. It moves the index forward by 7 code points, effectively pointing to the character `'W'` in the string.
+
+```java
+class OffsetByCodePoints {
+  public static void main(String[] args) {
+  String str = "Hello, World!";
+  int startIndex = 0;
+  int offsetIndex = 7;
+
+  int newIndex = str.offsetByCodePoints(startIndex, offsetIndex);
+  System.out.println("New Index: " + newIndex);
+  }
+}
+```
+
+The output would be:
+
+```
+New Index: 7
+```
+
+## Example 2
```

Comment:
You haven't addressed my issue with the duplicative examples:

>The next two examples are essentially the same as the first one. It might be good to show an example of fetching a character from the middle of a string of completely non-Latin script.

Reply:
@SSwiniarski  In Example 3, I have added a completely non-latin script.